notebooks/sequence_and_schema.ipynb

```diff
@@ -0,0 +1,172 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Data sequence and schema diagrams\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Data Sequences\n",
+    "### File apis\n",
+    "\n",
+    "```mermaid\n",
+    "---\n",
+    "title: Data sequence - POST /upload\n",
+    "---\n",
+    "\n",
+    "sequenceDiagram\n",
+    "    Django->>S3: file key, content\n",
+    "    Django->>Core: file key\n",
+    "    Core->>Workers: file key\n",
+    "    Core->>Elastic: file key\n",
+    "    S3->>Workers: file content\n",
+    "    Workers->>Elastic: chunk key, content\n",
+    "```\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Chat APIs\n",
+    "\n",
+    "\n",
+    "```mermaid\n",
+    "---\n",
+    "title: Data sequence - POST /chat/vanilla\n",
+    "---\n",
+    "\n",
+    "sequenceDiagram\n",
+    "    Django->>Core: ChatHistory.messages[]\n",
+    "    Core->>LLM API: ChatHistory.messages[]\n",
+    "\n",
+    "```\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```mermaid\n",
+    "---\n",
+    "title: Data sequence - POST /chat/rag\n",
+    "---\n",
+    "\n",
+    "sequenceDiagram\n",
+    "    Django->> Core: ChatHistory.messages[], File[].uuid\n",
+    "    Core->>Embedding API: ChatHistory.messages[]\n",
```

Comment:
what is the "Embedding API" here? i.e. is it a service?

Reply:
The embedding of user queries (i.e. the hot path as opposed to the async cold path used for chunks) happens inside of Core API. May be if we rename it `Core - Embedding`?

Reply:
this makes sense 👍

Reply:
maybe we take out this line and the subsequent one as it's all happening inside Core?

Reply:
That works